### DISCUSSION TITLE
Create cosmos db in another region, but in same deployment

### DISCUSSION THREAD

**Discussion by nikodemrafalski** [CompositionalIT/farmer#943] [Q&A] [answered]
Hi! 
Is it possible to create one resource in location other than owning resource group?

I have a bunch of resources in West Europe, but need to create cosmos db in another region.

At the moment I'm using this workaround : 

```fs
 arm {
        location envSettings.PrimaryLocation
        add_resources [virtualNetwork; privateDnsZone]
        add_resources [webAppsIdentity]
        add_resources [persistentStorageAccount; temporaryStorageAccount; versionedStorageAccount; kv]
        add_resources [apisAi]
        add_resources [
            resourceGroup {
                name "[resourceGroup().name]"
                deployment_name "cosmosdb_canadaeast"
                location Location.CanadaEast
                add_resource cosmodb
            }
        ]
        ...
        ...
        add_tags envSettings.Tags
    }
```
But it creates a nested deployment, so other resources cannot use `cosmodb` config to for example add its connection string to webapp config variables. and I'm getting error on live subscription "cosmosdb is not part of deployment"

**Comment by isaacabraham** [comment]
You could do it as a two step deployment i.e. first deploy the CosmosDB resource, and then a second deployment that uses the ResourceId of the CosmosDB instance as a "weak" link.

@TheRSP I know that you worked on a lot of the resource group deployment stuff - any thoughts here?

**Comment by r30e** [comment, marked as answer]
Personally, I've got around this issue with an extension on the deployment builder. Something like this:

```fsharp
[<CustomOperation "add_resources_in_location">]
member this.AddInLocation (state, location, resources) =
  this.AddArmResources(state, resources |> List.collect (fun x -> x.BuildResources location))
```
(Apologies for code quality, I'm writing on a phone)

However given that farmer bakes-in the location of a resource as part of the add_resources call, you can change the location of the resource group and this won't affect any resource which has already been added.

```fsharp
arm {
  location EastUS
  add_resources eastUsResources
  location CanadaEast
  add_resources canadaEastResources
}
```

**Comment by isaacabraham** [comment]
@TheRSP feel free to create that as a PR etc. - no problems with that from my side (although I would recommend documenting it as well to explain the use case for it).
